### ISSUE TITLE
quarto-1.4.80-macos unable to recognize the content of the "block"  in div when compiling beamer.

### ISSUE THREAD

**Issue by williamlorder** [quarto-dev/quarto-cli#5617]
### Bug description

A very frustrating problem, which did not exist in my previous version.

I want to input the Block block in beamer as follows in QMD:

:: {.block}
content
:::

So that it can be displayed as a block in be after compilation. This was successfully done in previous versions, but cannot be achieved in the latest(quarto-1.4.80-macos). After compilation, only content is displayed instead of the block.

Has this feature been removed from new version or is it just a bug?

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.
- [X] Please provide the output of `quarto check` so we know which version of quarto and its dependencies you're running.

**Comment by mcanouil** [comment]
Thanks for the report!

Could you share a small "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Comment by williamlorder** [comment]
Sure, mcanouil ! Here is a simple reproduction example
---md
title: "XXXX"
author: 
  - "XXX"
institute: "XXXX"
date: "2023/5/22"
format:
  beamer:
    aspectratio: 1610 
    navigation: horizontal #
    theme: "Boadilla"
    colortheme: "beaver"
    # fonttheme: "structurebold"
    # fontfamily: "arev"
    toc: true
    # number-sections: true
    # documentclass: ctexbeamer
    colorlinks: true 
    slide-level: 2
    # toc: true   
pdf-engine: xelatex
# bibliography: "/Users/zekai/References.bib"
---

# Section 1

## Subsection 1

:::{.block}
BLOCK EXAMPLE
:::

```

What I expect is
<img width="752" alt="截屏2023-05-21 21 47 22" src="https://github.com/quarto-dev/quarto-cli/assets/83387569/36b2e240-84b3-42d9-a03d-8b391ed6fa26">
(I used latex syntax to generate the block like 
```tex
\begin{block}
BLOCK EXAMPLE
\end{block}
```
is worked(use tex syntax in quarto). But failed when use div syntax which is valid for a certain version(I can't quite remember it, maybe some version from the previous year)

)


It came with the following screenshot
<img width="1004" alt="截屏2023-05-21 21 44 48" src="https://github.com/quarto-dev/quarto-cli/assets/83387569/48826b5b-c7ef-4823-94ee-60caba79a39d">

**Comment by dragonstyle** [comment]
I'm not aware of any versions of Quarto natively behaving the way you describe - there is an extension you can install which adds support for this behavior:

https://github.com/quarto-ext/latex-environment

HTH!

**Comment by cderv** [comment]
@williamlorder the `block` fenced div is an "undocumented" Pandoc feature, but it requires to set a Block title using a nested header - here level 3 `###`

Try this 

````markdown
---
title: "XXXX"
author: 
  - "XXX"
institute: "XXXX"
date: "2023/5/22"
format:
  beamer:
    aspectratio: 1610 
    navigation: horizontal #
    theme: "Boadilla"
    colortheme: "beaver"
    toc: true
    colorlinks: true 
    slide-level: 2
pdf-engine: xelatex
---

# Section 1

## Subsection 1

:::{.block}
### Block title
BLOCK EXAMPLE
:::

````

I say undocumented because https://github.com/jgm/pandoc/issues/7937 has the detail only

it is a behavior not clear from their doc at https://pandoc.org/MANUAL.html#structuring-the-slide-show

> Headings below the slide level in the hierarchy create headings within a slide. (In beamer, a “block” will be created. If the heading has the class example, an exampleblock environment will be used; if it has the class alert, an alertblock will be used; otherwise a regular block will be used.)

This means that any title below slide level will create a block.

So this is Pandoc behavior mainly, and I believe it still works with Quarto. 

> A very frustrating problem, which did not exist in my previous version.

Were you using block in fenced div without having a title inside ?

**Comment by dragonstyle** [comment]
> the block fenced div is an "undocumented" Pandoc feature

TIL :)

**Comment by mcanouil** [comment]
There are a lot of undocumented "features" in Pandoc[^1]. 😅

[^1]: are they really features or unintended behaviours. 😏

**Comment by williamlorder** [comment]
> @williamlorder the `block` fenced div is an "undocumented" Pandoc feature, but it requires to set a Block title using a nested header - here level 3 `###`
> 
> Try this
> 
> ```gfm
> ---
> title: "XXXX"
> author: 
>   - "XXX"
> institute: "XXXX"
> date: "2023/5/22"
> format:
>   beamer:
>     aspectratio: 1610 
>     navigation: horizontal #
>     theme: "Boadilla"
>     colortheme: "beaver"
>     toc: true
>     colorlinks: true 
>     slide-level: 2
> pdf-engine: xelatex
> ---
> 
> # Section 1
> 
> ## Subsection 1
> 
> :::{.block}
> ### Block title
> BLOCK EXAMPLE
> :::
> ```
> 
> I say undocumented because [jgm/pandoc#7937](https://github.com/jgm/pandoc/issues/7937) has the detail only
> 
> it is a behavior not clear from their doc at https://pandoc.org/MANUAL.html#structuring-the-slide-show
> 
> > Headings below the slide level in the hierarchy create headings within a slide. (In beamer, a “block” will be created. If the heading has the class example, an exampleblock environment will be used; if it has the class alert, an alertblock will be used; otherwise a regular block will be used.)
> 
> This means that any title below slide level will create a block.
> 
> So this is Pandoc behavior mainly, and I believe it still works with Quarto.
> 
> > A very frustrating problem, which did not exist in my previous version.
> 
> Were you using block in fenced div without having a title inside ?

You are right, After using `###` title you mentioned,it works! 

**The documentation of Pandoc is a mess. It's hard to find anything of reference value.**

**Comment by williamlorder** [comment]
Can this section added to Quarto's Beamer documentation? I believe the `Block` feature is important for Beamer presentation.I've summarized various Blocks are presented in pandoc below.

```markdown
:::{.block}
### Simple block
:::

:::{.alert}
### Alert block

:::

:::{.example}
### Example block

:::

```

Thanks!

**Comment by cderv** [comment]
Yes we could add it to https://quarto.org/docs/presentations/beamer.html probably. 

Though another usage following Pandoc's doc
> Headings below the slide level in the hierarchy create headings within a slide. (In beamer, a “block” will be created. If the heading has the class example, an exampleblock environment will be used; if it has the class alert, an alertblock will be used; otherwise a regular block will be used.

would be to write it like this 
```markdown
---
title: Test
format: beamer
keep-tex: true
slide-level: 2
---

## Slide

### Simple block

Content

### Alert block {.alert}

Content

### Example block {.example}

Content
```

Just adding for reference when we document it

**Comment by allenmanning** [comment]
@cwickham For consideration to add to our documentation.

**Comment by cderv** [comment]
@tarleb do you know what is exactly the pandoc supported feature for this, specific to beamer ? 

https://github.com/jgm/pandoc/issues/7937 is still open and it would nice to have this documented better in Pandoc so that if we document we rely on Pandoc's expected behavior and not possible side effect. 

For example, I believe this related to this part in the doc: https://pandoc.org/MANUAL.html#frame-attributes-in-beamer
and maybe example should be added there ? 

Just to confirm if / when `block` syntax is to be used compared to slide attributes uses classes on headers. 

Thanks !

**Comment by cwickham** [comment]
@cderv I have added your example to the Beamer docs with https://github.com/quarto-dev/quarto-web/pull/1206

I didn't add the `::: {.block}` example since we don't have confirmation this is stable/intended behavior. I'll leave this issue open, but move it to Future for that case.
